torch/testing/_internal/common_methods_invocations.py

```diff
@@ -14818,18 +14818,22 @@ def error_inputs_mean(op_info, device, **kwargs):
                          active_if=(IS_MACOS or IS_WINDOWS)),
         ),
     ),
-    OpInfo(
+    UnaryUfuncInfo(
         'nn.functional.threshold',
-        aten_backward_name='threshold_backward',
         ref=lambda x, threshold, value: np.where(x > threshold, x, value).astype(x.dtype),
```

Comment:
@mruberry @ngimel  I'm not sure what to aim for here.

1) my `UnaryUfuncInfo` preserves the existing `ref= ... np.where(x > threshold, ...` 
2) my _refs.nn.functional.threshold op seems to match the numpy ref
3) torch.nn.functional.threshold doesn't match either the numpy ref or my ref for nan handling

So.. should I be treating torch.nn.functional.threshold as the gold standard and perhaps update the numpy ref to match it? (If so i am not sure why the orig Opinfo test didn't catch this: oh- i guess by making a UnaryUfuncInfo I am opting into new test coverage for the first time?)

This makes a little more sense after i wrote it out, but i'm still not sure what the best change would be.

Reply:
I'm confused, to me it looks like torch.nn.functional.threshold and Will's decomposition produce the same result (numpy ref might be different):
```
In [2]: inf=float('inf')

In [3]: nan=float('nan')

In [4]: a=torch.tensor([inf, inf, inf, -inf, -inf, -inf, nan, nan, nan])

In [5]: threshold=-0.123

In [6]: value=9.

In [7]: torch.nn.functional.threshold(a, threshold, value)
Out[7]: tensor([inf, inf, inf, 9., 9., 9., nan, nan, nan])

In [9]: torch.where(a <= threshold, value, a)
Out[9]: tensor([inf, inf, inf, 9., 9., 9., nan, nan, nan])
```

Reply:
> I'm confused, to me it looks like torch.nn.functional.threshold and Will's decomposition produce the same result (numpy ref might be different):

yes, i agree with this @ngimel 

>my _refs.nn.functional.threshold op seems to match the numpy ref
torch.nn.functional.threshold doesn't match either the numpy ref or my ref for nan handling

in my statement above, i had modified my _refs version to match the numpy ref, hence the different conclusion.

It sounds like the numpy ref is the one that needs to change, so i'll try to get that to match nn.functional first and then see how my _refs version looks.